examples/03-javascripts-objects-classes/mars_rover.js

```diff
@@ -14,6 +14,81 @@
 
 class MarsRover {
 
+  constructor(x, y) {
+    this.positionX = x;
+    this.positionY = y;
+    this.direction = 'North';
+  }
+
+  position() {
+    return (
+      {
+        x: this.positionX,
+        y: this.positionY,
+        direction: this.direction
+      }
+    )
+  }
+
+  move() {
+    if (this.direction == 'North') {
+      this.positionX = this.position().x;
+      this.positionY = this.position().y +  1;
+    } else if (this.direction == 'Left') {
+      this.positionX = this.position().x -  1;
+      this.positionY = this.position().y;
+    } else if (this.direction == 'Right') {
+      this.positionX = this.position().x +  1;
+      this.positionY = this.position().y;
+    } 
+    return (
```

Comment:
@anneilustre Good initiative on returning the current coordinates on the move call. If you prefer to do this, I suggest move this code into a separate private method, and then just call it inside the methods you want to. This would reduce repetitions in the code.

```js
      {
        positionX: this.positionX, 
        positionY: this.positionY,
        direction: this.direction
      }
```